patches/llvm-HEAD.patch

```diff
@@ -618,3 +618,35 @@ index ac57360cca9e..c8edc8bc8f21 100644
  
  template <class _A1>
  inline _LIBCPP_HIDE_FROM_ABI
+diff --git a/runtimes/CMakeLists.txt b/runtimes/CMakeLists.txt
+index af1e3dbb8223..77f7421233c7 100644
+--- a/runtimes/CMakeLists.txt
++++ b/runtimes/CMakeLists.txt
+@@ -1,7 +1,27 @@
+ # This file handles building LLVM runtime sub-projects.
+ cmake_minimum_required(VERSION 3.13.4)
++
+ project(Runtimes C CXX ASM)
+ 
++# The CMake documentation on the APPLE variable reads:
++#   Set to True when the target system is an Apple platform (macOS, iOS, tvOS or
++#   watchOS).
++#
++# So this variable is used as a Target, however on Apple host platforms it is
++# automatically set to true, and it seems you can't change it through say the
++# command line your generator doesn't support platform specification, which
```

Comment:
This surprises me. What happens if in LLVM-embedded-toolchain-for-Arm/CMakeLists.txt you add -DAPPLE=False to the CMAKE_ARGS for runtimes?

Reply:
Yes I tried that. It doesn't register. It also doesn't register if in the runtimes CMakeLists.txt file I set APPLE to False before `project(Runtimes C CXX ASM)`, so I'm assuming that it's set somewhere in there.

I'm by far a CMake expert, and after I googled around for far too long, the best I got was setting the CMAKE_GENERATOR_PLATFORM variable. But as I noted in the code comment, this doesn't work for ninja as it doesn't support setting the platform specification.

It does baffle me and I'm assuming/hoping I missed something obvious, because why wouldn't you be able to change a variable that according to the documentation indicates which platform you're building for.

Reply:
ping

Reply:
I guess that makes sense since it's a variable that CMake defines. However modifying CMake internal state might be asking for trouble. Since this change isn't urgent I'd prefer we take longer and find a non-hacky fix.

Reply:
I agree, this actually looks like a candidate for https://cmake.org/cmake/help/book/mastering-cmake/chapter/Cross%20Compiling%20With%20CMake.html - do we need to specify the toolchain and maybe even reuse it (BMT toolchain definition) on other platforms for building libraries?

Reply:
I actually read that page, and tried to fix the issue with suggestions therein, but nothing helped. You can set `CMAKE_SYSTEM_NAME` to Generic, but then CMAKE will fail checks at least for embedded targets as it does certain LLVM checks that need host system support. And it might still set APPLE to True. Didn't check that.

I'm not arguing that we shouldn't solve this in a non-hackish way, but my guess is that this needs a saner setup on the LLVM side, which would need a bit more effort.

Reply:
> You can set `CMAKE_SYSTEM_NAME` to Generic, but then CMAKE will fail checks at least for embedded targets

Should be fixed by https://reviews.llvm.org/D144757. Review please!